### PR TITLE
Improve forgot password screen

### PR DESCRIPTION
**Pull request by Viacheslav83**

### CODE REVIEW

File: INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift [hunk 1]
```diff
@@ -13,67 +13,151 @@ class ForgotPasswordViewController: UIViewController, Storyboarded {
...
 
     }
     
-    func setupEmailTextField() {
+    override func animateKeyboardAppearance(height: CGFloat) {
+        
+        let heightView = UIScreen.main.bounds.height
+        let differentHeight = heightView - heightTableView
+        
+        var resultHeightConstreint: CGFloat
```
Comment (svitlana-korostelova) [inline INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift:54]: constraint

File: INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift [hunk 2]
```diff
@@ -13,67 +13,151 @@ class ForgotPasswordViewController: UIViewController, Storyboarded {
...
+            guard let emailCell = tableView.dequeueReusableCell(withIdentifier: "EmailTableViewCell") as? EmailTableViewCell else { return UITableViewCell() }
+            emailCell.textField.placeholder = "inputEmail".localized
+            emailCell.textField.textContentType = .emailAddress
+            return emailCell
+        case 8:
+            guard let buttonCell = tableView.dequeueReusableCell(withIdentifier: "SendButtonTableViewCell") as? SendButtonTableViewCell else { return UITableViewCell() }
+            buttonCell.delegate = self
+            return buttonCell
+        default:
+            return UITableViewCell()
```
Comment (svitlana-korostelova) [inline INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift:120]: Please refactor it like this: declare let cell before the switch, delete all returns and assign a value to a cell at the end of each case, in default case return cell ?? UITableViewCell()

File: INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift [hunk 3]
```diff
@@ -13,67 +13,151 @@ class ForgotPasswordViewController: UIViewController, Storyboarded {
...
-        emailTextField.borderStyle = .line
-        emailTextField.keyboardAppearance = .dark
-        emailTextField.backgroundColor = .white
-        emailTextField.keyboardType = .emailAddress
-        emailTextField.placeholder = "Enter your Email"
-        emailTextField.clearButtonMode = .whileEditing
+        forgotTableView.beginUpdates()
+        tableViewTopConstraint.constant = resultHeightConstreint
+        view.layoutSubviews()
+        forgotTableView.endUpdates()
```
Comment (spanastasia) [inline INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift:65]: scroll tableView to submit button using forgotTableView.scrollToRow(at: IndexPath(row: submitButtonRow, section: 0), at: .bottom, animated: true)

**Review by spanastasia — COMMENTED** [review]
Let's make the pushing view controller in ForgotPasswordCoordinator animated.

File: INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift [hunk 4]
```diff
@@ -12,68 +12,146 @@ class ForgotPasswordViewController: UIViewController, Storyboarded {
...
+        default:
+            return UITableViewCell()
+        }
+        return cell ?? UITableViewCell()
+    }
+    
+    func tableView(_ tableView: UITableView,
+                   heightForRowAt indexPath: IndexPath) -> CGFloat {
+        
+        var heightCell: CGFloat?
```
Comment (spanastasia) [inline INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift:118]: heightCell may be non-optional

File: INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift [hunk 5]
```diff
@@ -12,68 +12,146 @@ class ForgotPasswordViewController: UIViewController, Storyboarded {
...
+        case 1:
+            heightCell = widthOfView * 0.253
+        case 2:
+            heightCell = widthOfView * 0.3
+        case 3:
+            heightCell = widthOfView * 0.264
+        case 4:
+            heightCell = widthOfView * 0.147
+        default:
+            heightCell = 0
```
Comment (spanastasia) [inline INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift:132]: Maybe better make heightCell depending on view height, not width?

**Comment by Stepan19897** [comment]
LogoForgot cell, PassRecovery cell and Explanation cell needs to be embedded into one cell.

**Review by oleksiikolomiietssnapp — APPROVED** [review]

File: INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift (hunk 5 above)
Comment (Viacheslav83) [reply INTITA/Modules/ForgotPassword/ForgotPasswordViewController.swift]: Some elements of the screen are proportional, so in my opinion it is better to scale them depending on the width.

**Pull request merged**